GeneratorInterface/Core/interface/ConcurrentGeneratorFilter.h

```diff
@@ -112,6 +121,12 @@ namespace edm {
     void initLumi(gen::GenStreamCache<HAD, DEC>* cache, LuminosityBlock const& index, EventSetup const& es) const;
     ParameterSet config_;
     mutable std::atomic<gen::GenStreamCache<HAD, DEC>*> useInLumi_{nullptr};
+    mutable std::atomic<unsigned long long> greatestNStreamEndLumis_{0};
+    mutable std::atomic<bool> streamEndRunComplete_{true};
+    // The next two data members are thread safe and can be safely mutable because
+    // they are only modified/read in globalBeginRun and globalBeginLuminosityBlock.
+    mutable unsigned long long nGlobalBeginRuns_{0};
+    mutable unsigned long long nInitializedInGlobalLumi_{0};
```

Comment:
Is this actually `nInitializedInGlobalLumiAfterNewRun_`?

Reply:
You're right. That is a more accurate name. I will change it.